src/sage/algebras/free_algebra_element.py

```diff
@@ -203,6 +203,55 @@ def _mul_(self, y):
                     del z_elt[key]
         return A._from_dict(z_elt)
 
+    def is_unit(self):
+        r"""
+        Return ``True`` if ``self`` is invertible.
+
+        EXAMPLES::
+
+            sage: A.<x, y, z> = FreeAlgebra(ZZ)
+            sage: A(-1).is_unit()
+            True
+            sage: A(2).is_unit()
+            False
+            sage: A(1 + x).is_unit()
+            False
+            sage: A.<x, y> = FreeAlgebra(QQ, degrees=(1,-1))
+            sage: A(x * y).is_unit()
+            False
+            sage: A(2).is_unit()
+            True
+        """
+        if self.is_zero() or len(self) > 1 or not list(self)[0][0].is_one():
+            return False
+        c = self.leading_coefficient()
+        return c.is_unit()
```

Comment:
```suggestion
        mc = self._monomial_coefficients
        if not mc or len(mc) > 1:
            return False
        m, c = list(mc.items())[0]
        if not m.is_one():
            return False
        return c.is_unit()
```
I think this will be more efficient.